ElectronicObserver/Window/Tools/SortieRecordViewer/SortieCostViewer/AirBaseCostCalculator.cs

```diff
@@ -0,0 +1,273 @@
+	{
+		if (SortieRecord.CalculatedSortieCost.TotalAirBaseSupplyCost is not null)
+		{
+			return SortieRecord.CalculatedSortieCost.TotalAirBaseSupplyCost;
+		}
+
+		SortieCostModel? airBaseSupplyCost = TryGetAirBaseSupplyCostFromDatabase(airBases);
+
+		airBaseSupplyCost ??= CalculateAirBaseSupplyCost();
+
+		SortieRecord.CalculatedSortieCost.TotalAirBaseSupplyCost = airBaseSupplyCost;
+		Db.Sorties.Update(SortieRecord);
+		Db.SaveChanges();
+
+		return SortieRecord.CalculatedSortieCost.TotalAirBaseSupplyCost;
+	}
+
+	private SortieCostModel CalculateAirBaseSupplyCost()
+	{
+		SortieDetails ??= ToolService.GenerateSortieDetailViewModel(Db, SortieRecord);
+
+		if (SortieDetails is null) return new();
+
+		int aircraftLoss = GetAircraftLossFromAirBattles(SortieDetails.Nodes);
+
+		if (SortieDetails.Nodes.Any(n => n.AirBaseRaid is not null))
+		{
+			aircraftLoss += GetAircraftLossFromRaid(SortieDetails);
+		}
+
+		return AirBaseResupplyCost(aircraftLoss);
+	}
+
+	/// <summary>
+	/// Number of aircraft lost due to the base getting damaged.
+	/// </summary>
+	private static int GetAircraftLossFromRaid(SortieDetailViewModel sortieDetails)
+	{
+		IEnumerable<int> aircraftBeforeSortie = sortieDetails.FleetsBeforeSortie.AirBases
+			.Zip(sortieDetails.StrikePoints, (ab, sp) => (AirBase: ab, StrikePoints: sp))
+			.Where(t => t.StrikePoints is not null)
+			.Select(t => t.AirBase)
+			.Select(a => a.Squadrons.Values.Sum(s => s.AircraftCurrent));
+
+		IEnumerable<int>? aircraftInBattle = sortieDetails.Nodes
+			.SelectMany(n => n.AllPhases)
+			.OfType<PhaseBaseAirAttack>()
+			.FirstOrDefault()
+			?.Units
+			.DistinctBy(u => u.AirBaseId)
+			.Select(u => u.Stage1FCount);
+
+		if (aircraftInBattle is null) return 0;
+
+		int aircraftLostInRaid = aircraftBeforeSortie
+			.Zip(aircraftInBattle, (s, b) => s - b)
+			.Sum();
+
+		// can't lose more than 4 aircraft per air base in a raid
+		// a squadron can't go under 1 aircraft as a result of an air raid
+		int maxAircraftLossInRaid = sortieDetails.FleetsBeforeSortie.AirBases
+			.Select(a => a.Squadrons.Values)
+			.Select(c => c.Sum(s => Math.Max(0, s.AircraftCurrent - 1)))
+			.Sum(c => Math.Min(4, c));
+
+		if (aircraftLostInRaid > maxAircraftLossInRaid)
+		{
+			// todo: log or something, more tests would be good
+			return 0;
+		}
+
+		return aircraftLostInRaid;
+	}
+
+	private static int GetAircraftLossFromAirBattles(IEnumerable<SortieNode> nodes) => nodes
+		.SelectMany(n => n.AllPhases)
+		.Sum(p => p switch
+		{
+			PhaseBaseAirRaid r => r.Stage1FLostcount + r.Stage2FLostcount,
+			PhaseBaseAirAttack r => r.Units
+				.Select((u, i) => (i % 2) switch
+				{
+					1 => u.Stage1FLostcount + u.Stage2FLostcount,
+					_ => 0,
+				})
+				.Sum(),
+
+			_ => 0,
+		});
+
+	private SortieCostModel? TryGetAirBaseSupplyCostFromDatabase(List<IBaseAirCorpsData> airBases)
+	{
+		if (TryGetAirBaseState(Db, Time) is not List<ApiAirBase> airBaseState)
```

Comment:
Again Resharper suggesting to use not null pattern

Reply:
Yeah, apparently I disabled this because it's stupid.
![image](https://github.com/ElectronicObserverEN/ElectronicObserver/assets/40002167/f1770ed9-dc8e-45c5-80e4-e9e3fd64a8a2)
![image](https://github.com/ElectronicObserverEN/ElectronicObserver/assets/40002167/5fe1b285-68b8-4715-9823-fa07c1f1b9e5)

Reply:
Changed the solution config so that this is a hint instead of a warning. I don't think it makes sense here.